symphony-bdk-core/src/main/java/com/symphony/bdk/core/service/message/MessageService.java

```diff
@@ -197,60 +190,73 @@ public V4Message send(@Nonnull V4Stream stream, @Nonnull String message) {
-    return executeAndRetry("send", () -> messagesApi.v4StreamSidMessageCreatePost(
-        streamId,
-        authSession.getSessionToken(),
-        authSession.getKeyManagerToken(),
-        message,
-        null,
-        null,
-        null,
-        null
-    ));
+    return this.send(streamId, Message.builder().content(message).build());
   }
 
   /**
    * Sends a message to the stream ID passed in parameter.
    *
-   * @param streamId    the ID of the stream to send the message to
-   * @param message     the message to send to the stream
+   * @param stream    the stream to send the message to
+   * @param message   the message to send to the stream
    * @return a {@link V4Message} object containing the details of the sent message
    * @see <a href="https://developers.symphony.com/restapi/reference#create-message-v4">Create Message v4</a>
    */
-  public V4Message send(@Nonnull String streamId, @Nonnull Message message) {
-    File tempFile = this.createTemporaryAttachmentFile(message.getAttachment());
-    try {
-      return executeAndRetry("send", () -> messagesApi.v4StreamSidMessageCreatePost(
-          streamId,
-          authSession.getSessionToken(),
-          authSession.getKeyManagerToken(),
-          message.getContent(),
-          message.getData(),
-          message.getVersion(),
-          tempFile,
-          null
-          )
-      );
-    } finally {
-      try {
-        Files.deleteIfExists(tempFile.toPath());
-      } catch (IOException ignored) {
-
-      }
-    }
+  public V4Message send(@Nonnull V4Stream stream, @Nonnull Message message) {
+    return this.send(stream.getStreamId(), message);
   }
 
   /**
    * Sends a message to the stream ID passed in parameter.
    *
-   * @param stream    the stream to send the message to
-   * @param message   the message to send to the stream
+   * @param streamId    the ID of the stream to send the message to
+   * @param message     the message to send to the stream
    * @return a {@link V4Message} object containing the details of the sent message
    * @see <a href="https://developers.symphony.com/restapi/reference#create-message-v4">Create Message v4</a>
    */
-  public V4Message send(@Nonnull V4Stream stream, @Nonnull Message message) {
-    return send(stream.getStreamId(), message);
+  public V4Message send(@Nonnull String streamId, @Nonnull Message message) {
+    return this.executeAndRetry("send", () ->
+        this.doSend(streamId, message, this.authSession.getSessionToken(), this.authSession.getKeyManagerToken())
+    );
+  }
+
+  /**
+   * The generated {@link MessagesApi#v4StreamSidMessageCreatePost(String, String, String, String, String, String, File, File)}
+   * does not allow to send multiple attachments as well as in-memory files, so we have to "manually" process this call.
+   */
+  private V4Message doSend(String streamId, Message message, String sessionToken, String keyManagerToken) throws ApiException {
```

Comment:
don't we need a test for that? (maybe a real one doing an upload with mock server?)

Reply:
I modified acceptance criteria for `APP-3136` to consider creating a dedicated test for this method

Reply:
Last time I worked with `mockserver`, I couldn't find anyway to test uploading file call.